src/replica/python/schema_migration.py

```diff
@@ -1,209 +1,123 @@
-
-        # apply all migration scripts in a loop
-        result = None
-        current = self.current_version()
-        while True:
-
-            if version is not None and current >= version:
-                _log.debug("current version (%s) is now same or newer as requested (%s)",
-                           current, version)
-                break
-
-            # find all migrations for current version
-            scripts = [s for s in self.scripts if s[0] == current]
-            if not scripts:
-                _log.debug("no migration scripts found for current version (%s)", current)
-                break
-
-            # if there are more than one final version use the latest or requested one
-            final_versions = set([s[1] for s in scripts])
-            final = max(final_versions)
-            if version is not None:
-                if version in final_versions:
-                    final = version
-
-            # only use scripts for that final version
-            scripts = sorted([s[2:4] for s in scripts if s[1] == final])
-
-            # apply all scripts
-            for script, path in scripts:
-                _log.info("--- Executing migration script %s", script)
-                if do_migrate:
-                    utils.loadSqlScript(self.engine, path)
-                    _log.info("+++ Script %s completed successfully", script)
-
-            # make sure that current version is updated in database
-            if do_migrate:
-                query = "UPDATE QMetadata SET value = {} WHERE metakey = 'version'".format(final)
-                self.engine.execute(query)
-
-                # read it back and compare with expected
-                current = self.current_version()
-                if current != final:
-                    raise RuntimeError("failed to update version number in database to {}, "
-                                       "current version is now {}".format(final, current))
-            else:
-                # pretend that we migrated to this new version
-                current = final
-
-            result = final
-
-        return result
-
-
-def make_migration_manager(name, engine, scripts_dir):
-    """Factory method for qservReplica schema migration manager
     This method is needed to support dynamic loading in `qserv-smig` script.
+
     Parameters
     ----------
     name : `str`
-        Module name, e.g. "replica"
-    engine : object
-        Sqlalchemy engine instance.
+        Module name, e.g. "admin"
+    connection : dbapi connection
+        Database connection instance.
     scripts_dir : `str`
         Path where migration scripts are located, this is system-level directory,
         per-module scripts are usually located in sub-directories.
+    set_initial_configuration : function or `None`
+        A function to be called to set the initial configuration of the repl database.
+        Will only be called if the schema is migrated from None to version 1. Optional.
```

Comment:
This idea on migrating from version `None` to version `1` is not consistent with the above-defined migration of `admin` from `None` of `0`? Or, I'm missing some important concept of the version `0`.

Reply:
Sorry, it's a typo, it should say "Will only be called if the schema is migrated from None" or maybe "Will only be called if the repl database is uninitialized.". Does that make more sense now? (🤞)

Reply:
Yes, it would make sense. Though, it may not be needed as I have a PR that has been reviewed and approved by @andy-slac. The ticket brings support for the `snig`-based schema migration of the Replication system's database. This should make your life easier. I'm still looking through the comments on the ticket to make some changes. However, it's very likely that the ticket will be merged into the `master` branch before yours. When you will be rebasing it then you may need to make some changes to the way you do schema migration for module `replica`.

Reply:
Sure. If you give me a heads up when you do the merge that would be great, for when we rebase this ticket branch onto master again after that.